service/controller/clusterapi/v30/resource/s3object/create.go

```diff
@@ -4,70 +4,56 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/meta"
 
 	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v30/controllercontext"
 )
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
-	s3ObjectToCreate, err := toBucketObjectState(createChange)
+	cr, err := meta.Accessor(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return microerror.Mask(err)
 	}
+	s3Object, err := toS3Object(createChange)
```

Comment:
The `s3object` resource is way simpler now as we only dispatch the S3 Put Object Input around.